exporters/trace/elasticsearch/src/main/java/io/opencensus/exporter/trace/elasticsearch/ElasticsearchTraceExporter.java

```diff
@@ -0,0 +1,243 @@
+import io.opencensus.trace.samplers.Samplers;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.nio.charset.Charset;
+import java.util.Collection;
+import java.util.List;
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.GuardedBy;
+import javax.xml.bind.DatatypeConverter;
+
+/**
+ * An OpenCensus span exporter implementation which exports data to Elasticsearch.
+ *
+ * <p>Example of usage:
+ *
+ * <pre>{@code
+ * ElasticsearchConfiguration elasticsearchConfiguration
+ *     = ElasticsearchConfiguration.builder().setAppName("sample-app")
+ *             .setElasticsearchUrl("http://localhost:9200)
+ *             .setElasticsearchIndex("index").setElasticsearchType("type").build();
+ *    ElasticsearchTraceExporter.createAndRegister(elasticsearchConfiguration);
+ *   ... // Do work.
+ * }</pre>
+ *
+ * @since 0.13
+ */
+public class ElasticsearchTraceExporter {
+
+  private static final String REGISTER_NAME = ElasticsearchTraceExporter.class.getName();
+  private static final String CONTENT_TYPE = "application/json";
+  private static final String REQUEST_METHOD = "POST";
+  private static final int CONNECTION_TIMEOUT = 6000;
+  private static final Object monitor = new Object();
+  private static final JsonParserUtil jsonParserUtil = new JsonParserUtil();
+  private static final Tracer tracer = Tracing.getTracer();
+  private static final Sampler probabilitySampler = Samplers.probabilitySampler(0.0001);
+
+  @GuardedBy("monitor")
+  @Nullable
+  private static Handler handler = null;
+
+  public ElasticsearchTraceExporter() {}
+
+  /**
+   * Creates and registers the ElasticsearchTraceExporter to the OpenCensus.
+   *
+   * @param elasticsearchConfiguration {@link ElasticsearchConfiguration}
+   * @throws MalformedURLException when the Elasticsearch url is invalid.
+   * @throws IllegalStateException when required parameters in the configuration are missing. eg:
+   *     Elasticsearch type and Elasticsearch index.
+   * @since 0.13
+   */
+  public static void createAndRegister(ElasticsearchConfiguration elasticsearchConfiguration)
+      throws Exception {
+    synchronized (monitor) {
+      checkState(handler == null, "Elasticsearch exporter already registered.");
+      checkState(
+          elasticsearchConfiguration.getElasticsearchIndex() != null,
+          "Elasticsearch Index not specified");
+      checkState(
+          elasticsearchConfiguration.getElasticsearchType() != null,
+          "Elasticsearch type not specified");
+      checkState(
+          elasticsearchConfiguration.getElasticsearchUrl() != null,
+          "Elasticsearch URL not specified");
+      registerElasticsearchHandler(new ElasticsearchTraceHandler(elasticsearchConfiguration));
+    }
+  }
+
+  private static void registerElasticsearchHandler(Handler elasticsearchHandler) {
```

Comment:
To fix this, the quick solution I can think of:
@SuppressWarnings("nullness")

@sebright any better solution? I think the checker is wrong here.

Reply:
I don't think that the checker looks at synchronization, so it doesn't know that "handler" can't be changed by another thread between lines 105 and 106.  I would register "elasticsearchHandler" instead of "handler" to make it clear that this method is always registering the input handler.

Reply:
oh ok, makes sense